src/main/kotlin/net/lumalyte/lg/infrastructure/persistence/guilds/GuildVaultRepositorySQLite.kt

```
@@ -214,6 +214,22 @@ class GuildVaultRepositorySQLite(private val storage: Storage<Database>) : Guild
         }
     }
 
+    override fun getTopGoldBalances(limit: Int): List<Pair<UUID, Long>> {
+        if (limit <= 0) return emptyList()
+        val sql = "SELECT guild_id, balance FROM vault_gold ORDER BY balance DESC LIMIT ?"
+
+        return try {
+            val results = storage.connection.getResults(sql, limit)
+            results.mapNotNull { row ->
+                val id = try { UUID.fromString(row.getString("guild_id")) } catch (e: Exception) { null }
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)